ddtrace/contrib/botocore/patch.py

```diff
@@ -334,20 +355,43 @@ def patched_api_call(original_func, instance, args, kwargs):
         if region_name is not None:
             span._set_str_tag("aws.region", region_name)
 
-        result = original_func(*args, **kwargs)
+        # set analytics sample rate
+        span.set_tag(ANALYTICS_SAMPLE_RATE_KEY, config.botocore.get_analytics_sample_rate())
 
-        response_meta = result.get("ResponseMetadata")
-        if response_meta:
-            if "HTTPStatusCode" in response_meta:
-                span.set_tag(http.STATUS_CODE, response_meta["HTTPStatusCode"])
+        try:
+            result = original_func(*args, **kwargs)
+            _set_response_metadata_tags(span, result)
+
+            return result
+        except botocore.exceptions.ClientError as e:
+            # `ClientError.response` contains the result, so we can still grab response metadata
+            _set_response_metadata_tags(span, e.response)
+
+            # Check if the user defined an ignore rule for this exception
+            client_error_ignores = (
+                config.botocore.client_error_ignores
+            )  # type: DefaultDict[str, Config._HTTPServerConfig]
+            if span.resource in client_error_ignores:
+                status_code = span.get_tag(http.STATUS_CODE)
+                if status_code:
+                    status_code = int(status_code)
+                    if client_error_ignores[span.resource].is_error_code(status_code):
```

Comment:
```suggestion
                    if client_error_ignores[span.resource].is_error_code(int(status_code)):
```
I don't think `mypy` will like redefining a variable unless it's made a `Union` explicitly?